File: sdk/src/wallet/bindings/nodejs/examples/migrate-stronghold-snapshot-v2-to-v3.js
const path = require('path')
require('dotenv').config({ path: path.resolve(__dirname, '.env') });
const { AccountManager, CoinType, migrateStrongholdSnapshotV2ToV3 } = require('@iota/wallet');

var accountManagerOptions = {
    storagePath: './alice-database',
    clientOptions: {
        nodes: [process.env.NODE_URL],
        localPow: true,
    },
    coinType: CoinType.Shimmer,
    secretManager: {
        Stronghold: {
            snapshotPath: "fixtures/v2.stronghold",
            password: "current_password",
        },
    },
};

try {
    new AccountManager(accountManagerOptions);
} catch (error) {
    console.error(error);
}

migrateStrongholdSnapshotV2ToV3("fixtures/v2.stronghold", "current_password", "fixtures/v3.stronghold", "new_password")

accountManagerOptions = {
    storagePath: './alice-database',
    clientOptions: {
        nodes: [process.env.NODE_URL],
        localPow: true,
    },
    coinType: CoinType.Shimmer,
    secretManager: {
        Stronghold: {
            snapshotPath: "fixtures/v3.stronghold",
            password: "new_password",
        },
    },
};

const manager = new AccountManager(accountManagerOptions);

File: sdk/src/wallet/bindings/python/examples/migrate-stronghold-snapshot-v2-to-v3.py
import iota_wallet
from iota_wallet import IotaWallet, StrongholdSecretManager

client_options = {
    'nodes': ['https://api.testnet.shimmer.network'],
}

coin_type = 4219

secret_manager = StrongholdSecretManager("examples/fixtures/v2.stronghold", "current_password")

try:
   wallet = IotaWallet('./alice-database', client_options, coin_type, secret_manager)
except ValueError as e:
    print(e)

iota_wallet.migrate_stronghold_snapshot_v2_to_v3("examples/fixtures/v2.stronghold", "current_password", "examples/fixtures/v3.stronghold", "new_password")

secret_manager = StrongholdSecretManager("examples/fixtures/v3.stronghold", "new_password")

wallet = IotaWallet('./alice-database', client_options, coin_type, secret_manager)

account = wallet.create_account('Alice')

print(account['publicAddresses'])